logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+		rc = sqlite3_step(stmt_blobs_get_total_filesize);
+		if (unlikely(rc != SQLITE_ROW)) fatal_sqlite3_err(rc, __LINE__);
+		p_file_info->blob_total_size = sqlite3_column_int64(stmt_blobs_get_total_filesize, 0);
+		// debug(D_LOGS_MANAG, "p_file_info->blob_total_size: %lld for: %s\n", p_file_info->blob_total_size, p_file_info->filename);
+		rc = sqlite3_reset(stmt_blobs_get_total_filesize);
+		if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+
+		// TODO: Can db_release_lock(p_file_info->db_mut) be moved before if(blob_filesize > p_file_info-> blob_max_size) ?
+		db_release_lock(p_file_info->db_mut);
+		uv_sleep(p_file_info->buff_flush_to_db_interval * MS_IN_SEC);
+	}
+}
+
+/**
+ * @brief Process the events of the uv_loop_t related to the DB API
+ */
+static void db_loop_run(void *arg){
+	UNUSED(arg);
+    uv_run(db_loop, UV_RUN_DEFAULT);
+}
+
+void db_set_main_dir(char *dir){
+	main_db_dir = dir;
+}
+
+void db_init() {
+	int rc = 0;
+    char *err_msg = 0;
+    uv_fs_t mkdir_req;
+    
+    db_loop = mallocz(sizeof(uv_loop_t));
+    rc = uv_loop_init(db_loop);
+    if (unlikely(rc)) fatal_libuv_err(rc, __LINE__);
+
+	main_db_path = mallocz(strlen(main_db_dir) + sizeof(MAIN_DB) + 1);
+	sprintf(main_db_path, "%s/" MAIN_DB, main_db_dir);
+
+	/* Create databases directory if it doesn't exist. */
+    rc = uv_fs_mkdir(db_loop, &mkdir_req, main_db_dir, 0775, NULL);
+    if(rc == 0) {
+    	info("DB directory created: %s", main_db_dir);
+    }
+    else if (rc == UV_EEXIST) {
+    	info("DB directory %s found", main_db_dir);
+    }
+    else {
+        error("DB mkdir() %s/ error: %s", main_db_dir, uv_strerror(rc));
+        fatal_sqlite3_err(rc, __LINE__);
+    }
+    uv_fs_req_cleanup(&mkdir_req);
+
+	/* Create or open main db */
+	rc = sqlite3_open(main_db_path, &main_db);
+    if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+    
+    /* Configure main database */
+    rc = sqlite3_exec(main_db,
+                      "PRAGMA auto_vacuum = INCREMENTAL;"
+                      "PRAGMA synchronous = 1;"
+                      "PRAGMA journal_mode = WAL;"
+                      "PRAGMA temp_store = MEMORY;"
+                      "PRAGMA foreign_keys = ON;",
+                      0, 0, &err_msg);
+    if (unlikely(rc != SQLITE_OK)) {
+        error("Failed to configure database");
+        error("SQL error: %s\n", err_msg);
+        sqlite3_free(err_msg);
+        fatal_sqlite3_err(rc, __LINE__);
+    } else {
+        info("%s configured successfully", MAIN_DB);
+    }
+    
+    /* Create new main DB LogCollections table if it doesn't exist */
+    rc = sqlite3_exec(main_db,
+                      "CREATE TABLE IF NOT EXISTS " MAIN_COLLECTIONS_TABLE "("
+                      "Id 					INTEGER 	PRIMARY KEY,"
+					  "Machine_GUID			TEXT		NOT NULL,"
+					  "Log_Source_Path		TEXT		NOT NULL,"
+					  "Type					INTEGER		NOT NULL,"
+                      "DB_Dir 				TEXT 		NOT NULL"
```

Comment:
A perfect table creation, you added control for all fields, I loved :heart: ! Congratulations :tada: !

Reply:
Thanks! :)